### PR TITLE
New get_transactions() mehod and type hints fix

### PR DESCRIPTION
**Pull request by Badiboy**
And some minor comments

### CODE REVIEW

File: tronpy/tron.py [hunk 1]
```diff
@@ -693,14 +699,13 @@ def get_solid_transaction_info(self, txn_id: str) -> dict:
...
-    def list_witnesses(self) -> list:
+    def list_witnesses(self) -> dict:
         """List all witnesses, including SR, SRP, and SRC."""
         # NOTE: visible parameter is ignored
         ret = self.provider.make_request("wallet/listwitnesses", {"visible": True})
         witnesses = ret.get("witnesses", [])
         for witness in witnesses:
             witness["address"] = keys.to_base58check_address(witness["address"])
-
         return ret
```
Comment (andelf) [inline tronpy/tron.py:709]: My bad. Should return `witnesses`

File: tronpy/tron.py [hunk 2]
```diff
@@ -821,5 +826,68 @@ def broadcast(self, txn: Transaction) -> dict:
...
+        if order_by is not None:
+            payload = {"order_by": order_by}
+        if min_timestamp is not None:
+            payload = {"min_timestamp": min_timestamp}
+        if max_timestamp is not None:
+            payload = {"max_timestamp": max_timestamp}
+        if search_internal is not None:
+            payload = {"search_internal": search_internal}
+        ret = self.provider.make_request(
+            "v1/accounts/{}/transactions".format(keys.to_base58check_address(addr)),
```
Comment (andelf) [inline tronpy/tron.py:887]: Haven't planed to support trongrid API.
Sorry.

**Comment by andelf** [comment]
Many thanks. This is a FullNode-only API implementation.
I haven't planed to support TronGrid APIs.

**Comment by andelf** [comment]
How about adding a `TronGridClient`?

**Comment by Badiboy** [comment]
> This is a FullNode-only API implementation.
I did not found any mention about it.

> I haven't planed to support TronGrid APIs.
Frankly speaking I'm very new to Tron, so I see no difference between FullNode API and TronGrid API. If look on methods and data structures - methods are simillar, data structures are common (for example - get_transactions, where all transactions have the same format as the one in get_trasnaction.

But you are the library owner. If you do not want to have grid support (definitely I see no reason why, but may be you see: I'm nood in Tron) - I'll make my own custom extension. I thought it it is better to extend public library. :)

**Comment by andelf** [comment]
TronGrid is a close-sourced API implementation and is maintained by the TronGrid team.
The URL prefix is `/v1/`.

Since it is close-sourced, I prefer not to have it mess up with FullNode APIs. 
In addition, TronGrid APIs have a different error handling logic(e.g. HTTP error code for rate limit).

Thank you all the same. You're right, support of trongrid is a required feature. It will be introduced in tronpy v0.3.

How do you think about the following:

- Add trongrid.py, so it can be accessed via`tronpy.trongrid.TronGrid` or
- Add TronGrid APIs to `Tron.trongrid` property.

**Comment by andelf** [comment]
cherry-picked your 63ec3925d4051b36cc458d2fea73004134b5bcb7 to fix type hints.😄

**Comment by Badiboy** [comment]
> How do you think about the following:
> - Add trongrid.py, so it can be accessed viatronpy.trongrid.TronGrid or
> - Add TronGrid APIs to Tron.trongrid property.

I have no special oppinion about it. As for now, I just need list of address transactions, so I'll get them anyway. :)

**Comment by andelf** [comment]
OK, I'll implement it first.

**Comment by Badiboy** [comment]
Absolutely not urgent. I made a custom request, so I'm fine now. :)

**Pull request closed without merging**